samples/dnn/inpainting.cpp

```diff
@@ -0,0 +1,219 @@
+        return -1;
+    }
+    double aspectRatio = static_cast<double>(image.rows) / static_cast<double>(image.cols);
+    imgWidth = min(image.rows, image.cols);
+    fontSize = min(fontSize, (stdSize*imgWidth)/stdImgSize);
+    fontWeight = min(fontWeight, (stdWeight*imgWidth)/stdImgSize);
+
+    maskGray = Mat::zeros(image.size(), CV_8U);
+
+    cout<<keyboard_shorcuts<<endl;
+    namedWindow("Draw Mask");
+    setMouseCallback("Draw Mask", drawMask);
+
+    const string label = "Draw the mask on the image. Press space bar when done ";
+
+    for(;;) {
+        Mat displayImage = image.clone();
+        Mat overlay = image.clone();
+
+        double alpha = 0.5;
+        Rect r = getTextSize(Size(), label, Point(), fontFace, fontSize, fontWeight);
+        r.height += 2 * fontSize; // padding
+        r.width += 10; // padding
+        rectangle(overlay, r, Scalar::all(255), FILLED);
+        addWeighted(overlay, alpha, displayImage, 1 - alpha, 0, displayImage);
+        putText(displayImage, label, Point(10, fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+        putText(displayImage, "Press 'i' to increase and 'd' to decrease brush size", Point(10, 2*fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+
+        displayImage.setTo(Scalar(255, 255, 255), maskGray > 0); // Highlight mask area
+        imshow("Draw Mask", displayImage);
+
+        int key = waitKey(1);
+        if (key == 'i') {
+            brush_size += 1;
+            cout << "Brush size increased to " << brush_size << endl;
+        } else if (key == 'd') {
+            brush_size = max(1, brush_size - 1);
+            cout << "Brush size decreased to " << brush_size << endl;
+        } else if (key == ' ') {
+            break;
+        } else if (key == 27){
+            return -1;
+        }
+    }
+    destroyAllWindows();
+    cout<<"Processing image..."<<endl;
+    Mat image_blob = blobFromImage(image, scale, Size(width, height), mean_v, swapRB, false);
+
+    Mat mask_blob;
+    mask_blob = blobFromImage(maskGray, 1.0, Size(width, height), Scalar(0), false, false);
+    mask_blob = (mask_blob > 0);
+    mask_blob.convertTo(mask_blob, CV_32F);
+    mask_blob = mask_blob/255.0;
+
+    net.setInput(image_blob, "image");
+    net.setInput(mask_blob, "mask");
+
+    Mat output = net.forward();
+
+    Mat output_transposed(3, &output.size[1], CV_32F, output.ptr<float>());
+
+    vector<Mat> channels;
+    for (int i = 0; i < 3; ++i) {
+        channels.push_back(Mat(output_transposed.size[1], output_transposed.size[2], CV_32F,
+                                    output_transposed.ptr<float>(i)));
+    }
+    Mat output_image;
+    merge(channels, output_image);
+    output_image.convertTo(output_image, CV_8U);
+
+    int h = static_cast<int>(width * aspectRatio);
+    resize(output_image, output_image, Size(width, h));
+
+    cout<<"Process completed!!!"<<endl;
+    imshow("Inpainted Output", output_image);
+    waitKey(0);
```

Comment:
there should be a loop: until user presses 'escape', he/she should be able to draw another mask on the same original image.